hazelcast/src/hazelcast/client/network.cpp

```diff
@@ -165,10 +165,8 @@ namespace hazelcast {
 
                 spi::impl::ClientExecutionServiceImpl::shutdown_thread_pool(executor_.get());
 
+                // release the guard so that the io thread can stop gracefully
                 io_guard_.reset();
-                io_context_->stop();
```

Comment:
Already scheduled ones should be completed. And this fix for that
But it feels like we are not stopping new ones to be accepted as well. Will this cause a leak on io_context_ ? May be it will continue to run after connection_manager is deleted ?